modules/charon-utils/src/main/java/org/wso2/charon3/utils/usermanager/InMemoryUserManager.java

```diff
@@ -95,25 +96,17 @@ public List<Object> listUsersWithGET(Node rootNode, int startIndex, int count, S
         }
     }
 
-    private List<Object> listUsers(Map<String, Boolean> requiredAttributes) {
-        List<Object> userList = new ArrayList<>();
-        userList.add(0);
-        //first item should contain the number of total results
+    private UsersGetResponse listUsers(Map<String, Boolean> requiredAttributes) throws CharonException {
+
+        List<User> userList = new ArrayList<>();
         for (Map.Entry<String, User> entry : inMemoryUserList.entrySet()) {
             userList.add(entry.getValue());
         }
-        userList.set(0, userList.size() - 1);
-        try {
-            return (List<Object>) CopyUtil.deepCopy(userList);
-        } catch (CharonException e) {
-            logger.error("Error in listing users");
-            return  null;
-        }
-
+        return (UsersGetResponse) CopyUtil.deepCopy(new UsersGetResponse(userList.size(), userList));
```

Comment:
I don't see any requirement to keep a copy of the returned response. Since we are not modifying the return object later shall we remove CopyUtil.deepCopy